### PR TITLE
fix(curriculum): cat photo app step 26 to 30

### LINKED ISSUE #56314: Update steps 26-30 of cat photo workshop to use specific assert methods where appropriate
We recently updated the guidelines for writing assertions here 
https://contribute.freecodecamp.org/how-to-work-on-coding-challenges/#writing-tests

For consistency and matching what is in the docs, we should slowly start to update the tests for the workshops and labs in the new frontend cert.

For this issue, just focus on updating the tests for steps 26-30 of the cat photo app workshop.
We don't want to have a large PR that updates a whole bunch of tests in a bunch of steps and makes it harder to review. 

Here is the link to the workshop:
https://github.com/freeCodeCamp/freeCodeCamp/tree/main/curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app

The tests should be updated, where appropriate, to use specific methods like `assert.equal`, `assert.exists`, `assert.match`, etc instead of just `assert`.

Here is the complete API reference 
https://www.chaijs.com/api/assert/#method_exists

Just focus on the new frontend cert and don't worry about updating the RWD cat photo app since that will become legacy anyway.

### PR DESCRIPTION
**Pull request by gagan-bhullar-tech**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56314

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698458e123b4f403936994d.md [hunk 1]
```diff
@@ -16,20 +16,21 @@ After the `figure` element, add another `h3` element with the text:
...
 
 ```js
-assert(
-  document.querySelectorAll('main > section')[1].lastElementChild.nodeName ===
-    'H3' && code.match(/<\/h3\>/g).length === 2
+assert.equal(
+  document.querySelectorAll('main > section')[1].lastElementChild.nodeName,
+    'H3'
 );
+assert.lengthOf(code.match(/<\/h3\>/g).length,  2);
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698458e123b4f403936994d.md:23]:
```suggestion
assert.lengthOf(code.match(/<\/h3\>/g),  2);
```

Check to make sure you don't have any dangling `.length`s.

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669845af34899240d9196258.md [hunk 2]
```diff
@@ -20,28 +20,28 @@ After the second `section` element's last `h3` element, add an ordered list with
...
 
 The three `li` elements should be nested inside the `ol` element.
 
 ```js
-assert(
+assert.lengthOf(
   [...document.querySelectorAll('li')].filter(
     (item) => item.parentNode.nodeName === 'OL'
-  ).length === 3
+  ).length, 3
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669845af34899240d9196258.md:44]:
```suggestion
  ), 3
```

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669845af34899240d9196258.md [hunk 3]
```diff
@@ -60,7 +60,7 @@ assert.deepStrictEqual(
 You should only have one `ol` element.
 
 ```js 
-assert([...document.querySelectorAll('ol')].length == 1);
+assert.lengthOf([...document.querySelectorAll('ol')].length, 1);
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669845af34899240d9196258.md:63]:
```suggestion
assert.lengthOf([...document.querySelectorAll('ol')], 1);
```

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984881117baa41da8bbde0.md [hunk 4]
```diff
@@ -14,19 +14,19 @@ After the ordered list, add another `figure` element.
 Your `figure` element should have an opening tag. Opening tags have this syntax: `<elementName>`.
 
 ```js
-assert(document.querySelectorAll('figure').length === 2);
+assert.lengthOf(document.querySelectorAll('figure').length, 2);
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984881117baa41da8bbde0.md:17]:
```suggestion
assert.lengthOf(document.querySelectorAll('figure'), 2);
```

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984881117baa41da8bbde0.md [hunk 5]
```diff
@@ -14,19 +14,19 @@ After the ordered list, add another `figure` element.
...
 ```
 
 Your `figure` element should have a closing tag. Closing tags have a `/` just after the `<` character.
 
 ```js
-assert(code.match(/<\/figure>/g).length === 2);
+assert.lengthOf(code.match(/<\/figure>/g).length, 2);
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984881117baa41da8bbde0.md:23]:
```suggestion
assert.lengthOf(code.match(/<\/figure>/g), 2);
```

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984905e95af5432faa28b1.md [hunk 6]
```diff
@@ -16,40 +16,39 @@ To improve the accessibility of the image you added, add an `alt` attribute with
...
 ```js
-assert(document.querySelectorAll('figure').length === 2);
+assert.lengthOf(document.querySelectorAll('figure').length, 2);
 ```
 
 Your `figure` element should have a closing tag. Closing tags have a `/` just after the `<` character.
 
 ```js
-assert(code.match(/<\/figure>/g).length === 2);
+assert.lengthOf(code.match(/<\/figure>/g).length, 2);
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984905e95af5432faa28b1.md:25]:
```suggestion
assert.lengthOf(code.match(/<\/figure>/g), 2);
```

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984905e95af5432faa28b1.md [hunk 7]
```diff
@@ -16,40 +16,39 @@ To improve the accessibility of the image you added, add an `alt` attribute with
 Your `figure` element should have an opening tag. Opening tags have this syntax: `<elementName>`.
 
 ```js
-assert(document.querySelectorAll('figure').length === 2);
+assert.lengthOf(document.querySelectorAll('figure').length, 2);
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984905e95af5432faa28b1.md:19]:
```suggestion
assert.lengthOf(document.querySelectorAll('figure'), 2);
```

**Comment by gagan-bhullar-tech** [comment]
@a2937 thanks. Updated PR.

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669845af34899240d9196258.md [hunk 8]
```diff
@@ -20,28 +20,28 @@ After the second `section` element's last `h3` element, add an ordered list with
...
 ```js
-assert(code.match(/<\/ol>/));
+assert.match(code, /<\/ol>/);
 ```
 
 The `ol` element should be above the second `section` element's closing tag. You have them in the wrong order.
 
 ```js
-assert($('main > section')[1].lastElementChild.nodeName === 'OL');
+assert.equal($('main > section')[1].lastElementChild.nodeName, 'OL');
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669845af34899240d9196258.md:35]: Hey buddy. Can you remove the remaining jQuery calls still present?

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984881117baa41da8bbde0.md [hunk 9]
```diff
@@ -14,19 +14,19 @@ After the ordered list, add another `figure` element.
...
 ```js
-assert(code.match(/<\/figure>/g).length === 2);
+assert.lengthOf(code.match(/<\/figure>/g), 2);
 ```
 
 There should be a `figure` element right above the second `section` element's closing tag.
 
 ```js
-assert($('main > section')[1].lastElementChild.nodeName === 'FIGURE');
+assert.equal($('main > section')[1].lastElementChild.nodeName, 'FIGURE');
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984881117baa41da8bbde0.md:29]: Remove the jQuery call here

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669848cd668e7242872f4861.md [hunk 10]
```diff
@@ -14,26 +14,26 @@ Inside the `figure` element you just added, nest an `img` element with a `src` a
...
 ```js
-assert(code.match(/<\/figure>/g).length >= 2);
+assert.isAtLeast(code.match(/<\/figure>/g).length, 2);
 ```
 
 There should be a second `figure` element right above the second `section` element's closing tag. You have them in the wrong order.
 
 ```js
-assert($('main > section')[1].lastElementChild.nodeName === 'FIGURE');
+assert.equal($('main > section')[1].lastElementChild.nodeName, 'FIGURE');
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669848cd668e7242872f4861.md:29]: And here

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984905e95af5432faa28b1.md [hunk 11]
```diff
@@ -16,40 +16,39 @@ To improve the accessibility of the image you added, add an `alt` attribute with
...
 ```js
-assert(code.match(/<\/figure>/g).length === 2);
+assert.lengthOf(code.match(/<\/figure>/g), 2);
 ```
 
 There should be a `figure` element right above the last `section` element's closing tag.
 
 ```js
-assert($('main > section')[1].lastElementChild.nodeName === 'FIGURE');
+assert.equal($('main > section')[1].lastElementChild.nodeName, 'FIGURE');
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984905e95af5432faa28b1.md:31]: And here

**Comment by gagan-bhullar-tech** [comment]
@a2937 updated PR.

**Review by majestic-owl448 — APPROVED** [review]
LGTM 👍

**Review by a2937 — APPROVED** [review]
Looks good to me.

**Pull request merged**